test: Use rpc_deprecated only for testing deprecation

fjahr opened the pull request:
The comment in `functional/rpc_deprecated.py` says "This test should be used to verify correct behaviour of deprecated RPC methods with and without the -deprecatedrpc flags." I think we can get rid of the "with" part since we can assume that every deprecated RPC is already tested in at least one other functional test. (I didn't look but I could verify in our coverage if someone has doubts about that.) In order for this test to continue working, the flag will need to be used there. Otherwise this seems to prescribe copy+pasting a basic test from another file and I don't see a good reason for that.

fjahr committed:
test: Use rpc_deprecated only for testing deprecation

polespinasa reviewed (commented):
Concept ACK.

No need to double test a functionality.

rkrux commented on test/functional/rpc_deprecated.py:
```diff
@@ -7,21 +7,19 @@
         # assert_raises_rpc_error(-32, 'The wallet generate rpc method is deprecated', self.nodes[0].rpc.generate, 1)
-        # self.generate(self.nodes[1], 1)
+        #
+        # Please ensure that the -depgrecatedrpc flag for this RPC method is
+        # used in anther test where the RPC functionality is actually tested.
```
Couple typos
`depgrecatedrpc`
`anther`

fjahr replied:
fixed

rkrux commented on test/functional/rpc_deprecated.py:
```diff
@@ -7,21 +7,19 @@
 
     def run_test(self):
         # This test should be used to verify correct behaviour of deprecated
-        # RPC methods with and without the -deprecatedrpc flags. For example:
+        # RPC methods without the -deprecatedrpc flag. For example:
```
This is the final comment after this change:

> This test should be used to verify correct behaviour of deprecated RPC methods without the -deprecatedrpc flag.

I think it can be reworded to be more explicit in the intention. I'm assuming the deprecated RPCs will just throw errors, please correct me if I'm wrong.

```
This test should be used to verify the errors of the currently deprecated RPC methods (without the -deprecatedrpc flag) until such RPCs are fully removed.
```

Suggesting this because first time I have looked at this test and I found it a little strange that it doesn't test any RPCs at the moment until I saw the file commit history.

fjahr replied:
Added, thanks!

rkrux commented on test/functional/rpc_deprecated.py:
```diff
@@ -7,21 +7,19 @@
+        # Please ensure that the -depgrecatedrpc flag for this RPC method is
+        # used in anther test where the RPC functionality is actually tested.
 
         self.log.info("No tested deprecated RPC methods")
```
Nit:
```
 self.log.info("No tested currently deprecated RPC methods")
```

fjahr replied:
Changed this but with slightly different wording

rkrux reviewed (commented):
Concept ACK c5e384de6c0b92692021272a7464cd26a54d298c

New to this test, suggested some verbiage change. PLMK if it doesn't fit.

fjahr commented:
Addressed the comments by @rkrux , thanks!

rkrux commented on test/functional/rpc_deprecated.py:
```diff
@@ -7,23 +7,22 @@
 class DeprecatedRpcTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.num_nodes = 2
+        self.num_nodes = 1
```
1 node fewer is an additional benefit.

rkrux commented on test/functional/rpc_deprecated.py:
```diff
@@ -7,23 +7,22 @@
-        # RPC methods with and without the -deprecatedrpc flags. For example:
+        # This test should be used to verify the errors of the currently
+        # deprecated RPC methods (without the -deprecatedrpc flag) until
+        # such RPCs are fully removed.
```
Missed the `For example` part. In case retouched, fine otherwise.

fjahr replied:
Added it back now

rkrux reviewed (approved):
> since we can assume that every RPC is already tested

In the PR description, it can be reworded to say `... that every deprecated RPC is ...` in order to narrow down its intent.

Thanks for addressing the comments, ACK b1c80fab0c026a40f2901958abdeedb9dd36c30d

polespinasa commented on test/functional/rpc_deprecated.py:
nit: update copyright

fjahr replied:
done

polespinasa reviewed (commented):
ACK b1c80fab0c026a40f2901958abdeedb9dd36c30d

janb84 commented on test/functional/rpc_deprecated.py:
```diff
@@ -7,23 +7,22 @@
         # assert_raises_rpc_error(-32, 'The wallet generate rpc method is deprecated', self.nodes[0].rpc.generate, 1)
-        # self.generate(self.nodes[1], 1)
+        #
+        # Please ensure that the -deprecatedrpc flag for this RPC method is
+        # used in another test where the RPC functionality is actually tested.
```
nit: Needed to read this comment several times before understanding, suggested small change for clarity. 
```suggestion
        # Please ensure that the -deprecatedrpc flag for all the RPC methods
        # mentioned above is used in their respective functional tests, 
        # where the RPC functionality is actually tested.
```

fjahr replied:
I have reworked that part a bit, I hope it's clearer now.

janb84 reviewed (commented):
Concept ACK [b1c80fa](https://github.com/bitcoin/bitcoin/commit/b1c80fab0c026a40f2901958abdeedb9dd36c30d)

Suggested small comment change for clarity, feel free to ignore.

musaHaruna reviewed (commented):
Concept ACK [b1c80fa](https://github.com/bitcoin/bitcoin/commit/b1c80fab0c026a40f2901958abdeedb9dd36c30d) 
New to this test, please correct me know if am wrong, after checking the history of the file, it seems that it has primarily been used to test deprecated RPCs both with and without the -deprecatedrpc flag. This change proposes to focus exclusively on testing the behavior of deprecated RPCs without the flag, as other tests already cover the cases where the -deprecatedrpc flag is used.

fjahr commented:
Addressed the comments now since there were a few that made sense to fix.

> New to this test, please correct me know if am wrong, after checking the history of the file, it seems that it has primarily been used to test deprecated RPCs both with and without the -deprecatedrpc flag. This change proposes to focus exclusively on testing the behavior of deprecated RPCs without the flag, as other tests already cover the cases where the -deprecatedrpc flag is used.

Yes, that's correct!

janb84 reviewed (commented):
re ACK [2819c51](https://github.com/bitcoin/bitcoin/commit/2819c514825577dc8d73690882f25117d412ed9f)
Thank you for rewriting the comment - the intention is much easier to understand now (IMO)

polespinasa reviewed (commented):
reACK 2819c514825577dc8d73690882f25117d412ed9f
